From d818dc50d82b14dc27bbd486db567e1567621e35 Mon Sep 17 00:00:00 2001
From: Nicholas Vinson <nvinson234@gmail.com>
Date: Fri, 12 Jan 2024 22:36:44 -0500
Subject: [PATCH] Restrict symbol version map to global symbols

ld.lld-17 fails to link when symbols such as

    ld.lld: error: version script assignment of 'SUNWprivate_1.1' to
    symbol '_ZTVZ21WB_HandshakeWalkStackE16TraceSelfClosure' failed:
    symbol not defined

    ld.lld: error: version script assignment of 'SUNWprivate_1.1' to
    symbol '_ZTVZ24WB_HandshakeReadMonitorsE19ReadMonitorsClosure'
    failed: symbol not defined

    ld.lld: error: version script assignment of 'SUNWprivate_1.1' to
    symbol '_ZTVZ26WB_AsyncHandshakeWalkStackE16TraceSelfClosure'
    failed: symbol not defined

are listed in the version map. nm indicates these symbols are defined as
local symbols. Remove them from the version map to resolve the issues.

Fixes Gentoo bug 922147 ( https://bugs.gentoo.org/922147 )

Signed-off-by: Nicholas Vinson <nvinson234@gmail.com>
---
 make/hotspot/lib/JvmMapfile.gmk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/make/hotspot/lib/JvmMapfile.gmk b/make/hotspot/lib/JvmMapfile.gmk
index 2808ac2af03..6a8f22454b0 100644
--- a/make/hotspot/lib/JvmMapfile.gmk
+++ b/make/hotspot/lib/JvmMapfile.gmk
@@ -53,7 +53,7 @@ endif
 # platform dependent.
 
 ifeq ($(call isTargetOs, linux), true)
-  DUMP_SYMBOLS_CMD := $(NM) $(NMFLAGS) --defined-only *$(OBJ_SUFFIX)
+  DUMP_SYMBOLS_CMD := $(NM) $(NMFLAGS) --defined-only --extern-only *$(OBJ_SUFFIX)
   ifneq ($(FILTER_SYMBOLS_PATTERN), )
     FILTER_SYMBOLS_PATTERN := $(FILTER_SYMBOLS_PATTERN)|
   endif
